fairscale/optim/oss.py

```diff
@@ -67,6 +81,12 @@ def __init__(self, params: _params_t, optim: Type[Optimizer] = SGD, group: Any =
         # Current device is set by the parameters allocated to this rank
         self._device = split_param_groups[self.rank][0]["params"][0].device
 
+        # Broadcast buffer settings
+        self._buffer: Optional[torch.Tensor] = None
```

Comment:
parameters cannot change after init so you could pre-process the params and pre-create the batch buffers.